fabric-fluid-v1/src/main/java/net/fabricmc/fabric/mixin/fluid/MixinBucketItem.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.mixin.fluid;
+
+import java.util.Optional;
+
+import org.spongepowered.asm.mixin.Final;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.fluid.Fluid;
+import net.minecraft.item.BucketItem;
+import net.minecraft.sound.SoundCategory;
+import net.minecraft.sound.SoundEvent;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.WorldAccess;
+import net.minecraft.world.event.GameEvent;
+
+import net.fabricmc.fabric.api.fluid.v1.FabricFlowableFluid;
+
+@Mixin(BucketItem.class)
+public class MixinBucketItem {
+	@Shadow
+	@Final
+	private Fluid fluid;
+
+	@Inject(method = "playEmptyingSound", at = @At("HEAD"), cancellable = true)
+	private void playEmptyingSound(PlayerEntity player, WorldAccess world, BlockPos pos, CallbackInfo ci) {
```

Comment:
This mixin can duplicate less of the vanilla code if you just `ModifyVariable` the sound event before the `playSound` call

Reply:
I'd say default to the water sound like vanilla. (So `return emptySound.orElse(existing);` in a modifyvariable for example) That seems adequate for most modded fluids since they typically behave like coloured water anyway and already use this sound.

For a no-sound bucket, the modder can just override `BucketItem.playEmptyingSound` to do nothing. I feel like that's good enough since I'd imagine most modders want their buckets to have sounds.

Reply:
i've done by maintaining the default sound into the "orElse" case, this is the more simply way to not change the default behaviour of existing fluids with the interface-injection feature